float64 distance
---
bool result
---
float64 current_distance
